src/views/List.jsx

```diff
@@ -1,15 +1,45 @@
 import { ListItem } from '../components';
+import { useState } from 'react';
 
 export function List({ data }) {
+	const [searchItem, setSearchItem] = useState('');
+
+	const handleTextChange = (event) => {
+		setSearchItem(event.target.value);
+	};
+
+	const filteredItems = data.filter((item) => {
+		return item.name.toLowerCase().includes(searchItem.toLowerCase());
+	});
+
 	return (
 		<>
-			<p>
-				Hello from the <code>/list</code> page!
-			</p>
+			<form onSubmit={(event) => event.preventDefault()}>
+				<label htmlFor="search-item">Search Item: </label>
+
+				<input
+					id="search=item"
+					type="text"
+					placeholder="Search Item..."
+					onChange={handleTextChange}
+					value={searchItem}
+				/>
+
+				{searchItem && (
+					<button
+						type="button"
+						onClick={() => {
+							setSearchItem('');
+						}}
+					>
+						X
+					</button>
```

Comment:
While this is a great way to clear the field, another way to simplify the code and leverage native HTML functionality would be to change the type on line 22 to `search`! This provides a built-in reset button that allows users to clear the input field!

Reply:
Oh this is a cool suggestion @ahsanatzapier. This didn't exist when I did HTML more often and I've never heard of it. 😆

[Here's more details in the docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/search).

Reply:
Great suggestion, I have implemented the change for line 22. As a result of this change, I now have two buttons.

<img width="399" alt="Screenshot 2024-09-01 at 12 10 04 PM" src="https://github.com/user-attachments/assets/57d12851-3104-4188-93d2-b67d9074e9d0">

When I increase scale to test for accessibility purposes, the default clear search button icon in the search field does not correspondingly adjust in size. This can be addressed by styling down the line. 

<img width="1327" alt="Screenshot 2024-09-01 at 12 11 08 PM" src="https://github.com/user-attachments/assets/487a0e7c-401d-4076-8086-0cf190979205">